src/servers/lobby/packets/in/trade/trade_add_card_event.hpp

```diff
@@ -0,0 +1,70 @@
+#ifndef TRADE_ADD_CARD_EVENT_HPP
+#define TRADE_ADD_CARD_EVENT_HPP
+
+#include <memory>
+#include "lobby_packet_event.hpp"
+#include "lobby_session.hpp"
+#include "trade_add_card_self_event.hpp"
+#include "trade_add_card_other_event.hpp"
+#include "trade.hpp"
+
+class TradeAddCardEvent : public LobbyPacketEvent {
+public:
+	TradeAddCardEvent() : LobbyPacketEvent() {};
+	void Read(LobbySession* session, ClientPacket& pack) override
+	{
+		uint32_t targetPlayerId = pack.ReadInt();
+		uint8_t state = pack.ReadByte(); // 100 = add, 101 = remove, 102? not sure
+		uint32_t unk = pack.ReadInt();
+
+		std::array<char, 40> cardInfo;
+		for (size_t i = 0; i < cardInfo.size(); ++i)
+		{
+			cardInfo[i] = pack.buffer[i + pack.bufferPosition];
+		}
```

Comment:
```suggestion
		std::array<char, 40> cardInfo = pack.ReadArray<char, 40>();
```

Is this not sufficient? 😛

Reply:
And I see what you're doing here. I will make some kind of validation for this because we don't know what item is being traded right here and now, and we don't even know if the player even HAS the item to trade, but this item will be directly given to the client.. AKA scams.

Reply:
Eh yeah, ReadArray should work just fine, didn't see that function implemented.
And good point, validation should be easy by using the database. You can also check if all the fields line up with what the server receives in the packet, the cardInfo is the same structure that is being used in the inventory.